### ISSUE TITLE
Relative links stop working after 1 cycle

### ISSUE THREAD

**Issue by VaveD** [miguel-perez/smoothState.js#200]
Demo: http://dirkvanveghel.nl/smoothbug/portfolio/

Steps to reproduce:
- Click on About in the menu.
- Click on Portfolio in the menu.
  (or the other way around, doesn't matter which page you start)

Expected result:
You navigate to the Portfolio page.

Actual result:
You navigate 'back' one directory too many, or just back to the root. 

Tested in Chrome and FF.

Description:
When you switch from Portfolio > About the first time it works, but when you want to navigate back to portfolio again via the same menu it fetches the URL wrong and goes back one directory too many it seems. When you refresh the bug is also gone, so I assume I didn't just code anything wrong. Sorry for my non-native English, or if I'm just doing something wrong.

**Comment by ghost** [comment]
I was having a similar issue where if I clicked on the link that navigates to the current page I would get a blank page with **2015** written to the top left page.

**Your current links look like this**:
`<a href="../about">About</a>`
     and 
`<a href="../portfolio">Portfolio</a>`

**Change them to**:
`<a href="http://dirkvanveghel.nl/smoothbug/about">About</a>` 
    and 
`<a href="http://dirkvanveghel.nl/smoothbug/about">Portfolio</a>`

Extending the `href` attribute seems to have solved my issue.

**Comment by ghost** [comment]
Though this does create issues for working on your site during development. Double edged sword.

**Comment by DigitalMachinist** [comment]
@samDonald Yeah, it's good that there's a workaround but being forced to supply absolute URLs can really shoot site portability in the foot when it comes to moving between domains or deploying a site for a client. I can definitely see value in fixing this, so I'll give it a shot.

That being said, I tried to set up an example similar to @VaveD's with no luck. I haven't been able to reproduce the issue. I would try to debug this on vave's example site, but since only minified JS is being used without sourcemaps, I can't breakpoint to figure out where the code might be going wrong, and of course I have no ability to log anything since I can't change the source code.

Would you be able to swap out your JS for unminified versions, @VaveD? It would definitely help me to see what's going wrong in your copy and diagnose the issue, rather than trying to reproduce it myself.

**Comment by VaveD** [comment]
@DigitalMachinist An extra slash at the end seems to do the job actually. So having the href being "../portfolio/" instead of "../portfolio" seems to work. Did you maybe do that in your reproduction, or is there still a difference between my minified version and the current full version on this matter?

**Comment by DigitalMachinist** [comment]
@VaveD My test is using no ending slash as well, so it doesn't seem to be that, although it's helpful to know that when the issue presents, it can be corrected by adding an ending slash. I had my suspicions that this had to do with some kind of bad URL concatenation, so that strengthens that theory.

I'm not really concerned that there's a behavioural difference between the minified and unminified files. Mostly I was just hoping to set up breakpoints in jquery.smoothState.js in Chrome while I view your site so I can inspect possible problem areas more granularly. Since you're using jquery.smoothState.min.js, I really only have the option to breakpoint 1 of the 2 lines of code, and that constitutes half of the script, so breakpoints become basically useless.

Of course, if we can track down the issue and resolve it, you should be able to update your smoothState version and go back to using jquery.smoothState.min.js just the same.

**Comment by alisonjoseph** [comment]
I'm running into the same issue where I need to use relative paths, and because of how my dev/production environments are setup I'm unable to use the same absolute paths on both. Has anyone come up with a fix for this? Or am I just stuck changing the paths when I push stuff live.
